mne/annotations.py

```diff
@@ -517,3 +517,119 @@ def _ensure_annotation_object(obj):
     if not isinstance(obj, Annotations):
         raise ValueError('Annotations must be an instance of '
                          'mne.Annotations. Got %s.' % obj)
+
+
+def _counter_factory():
```

Comment:
We don't use this `_factory` anywhere in MNE, you can just do `class Counter` and instantiate it with `Counter()` rather than using `_counter_factory()` everywhere

Reply:
you are right. The only thing I don't like is that `Counter()` is not a function anymore, but an object. And it cannot be used as an example. But there's `_constant_id` in the tests and `_strip_blabla` in `eeglab.py`.

So I guess that is clear enough, and the code gets simplified.